go/vt/tabletserver/sysloglogger/sysloglogger_test.go

```diff
@@ -0,0 +1,62 @@
+package sysloglogger
+
+import (
+	"log/syslog"
+	"testing"
+	"time"
+
+	"github.com/youtube/vitess/go/vt/tabletserver"
+)
+
+// fakeWriter is a mock, implementing the syslogWriter interface, which can be used in place of
+// the real syslog writer to enable capturing and playing back of log messages in unit testing.
+type fakeWriter struct {
+	priority syslog.Priority
+	message  string
+	err      error // if non-nil, force an error to be returned
+}
+
+func (fw *fakeWriter) write(pri syslog.Priority, msg string) error {
+	if fw.err != nil {
+		return fw.err
+	}
+
+	fw.priority = pri
+	fw.message = msg
+	return nil
+}
+
+func (fw *fakeWriter) Info(msg string) error { return fw.write(syslog.LOG_INFO, msg) }
+func (fw *fakeWriter) Close() error          { return nil }
+
+func TestSyslog(t *testing.T) {
+	// Overwrite the usual syslog writer and StatsLogger subscription channel with mocks
+	mock := &fakeWriter{}
+	writer = mock
+	ch = make(chan interface{})
+
+	// Start running the plugin loop
+	go run()
+
+	// Send a fake message to the mock channel, and close it to end the plugin loop
+	ch <- &tabletserver.LogStats{
```

Comment:
Let's just make tabletserver.newLogStats public (replacing it with NewLogStats) and call it here, with context.Background().